openapi/access.yaml

```diff
@@ -373,10 +373,82 @@ paths:
                   value:
                     type: string
                     format: byte
-        400:
+        '400':
           $ref: '#/components/responses/400BadRequest'
-        500:
+        '500':
           $ref: '#/components/responses/500InternalServerError'
+  /status:
+    get:
+      summary: Status of the network
+      tags:
+        - Status
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Status'
+        '404':
+          $ref: '#/components/responses/404NotFound'
+        '500':
+          $ref: '#/components/responses/500InternalServerError'
+      operationId: GetNetworkStatus
+  /events:
+    summary: Query on-chain events in specified blocks. 
+    get:
+      tags:
+        - Events
+      responses:
+        '200':
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/BlockEvents'
+          description: OK
+        '400':
+          $ref: '#/components/responses/400BadRequest'
+        '404':
+          $ref: '#/components/responses/404NotFound'
+        '500':
+          $ref: '#/components/responses/500InternalServerError'
+      parameters:
+        - name: type
+          in: query
+          description: The event type is name-spaced with the address of the account and contract in which the event is declared.
+          required: true
+          deprecated: false
+          schema:
+            $ref: '#/components/schemas/EventType'
+          allowEmptyValue: false
+        - name: start_height
+          in: query
+          required: false
+          description: The start height of the block range for events. Must be used together with `end_height`. This parameter is incompatible with `block_ids`.
+          schema:
+            $ref: '#/components/schemas/BlockHeight'
+          allowEmptyValue: false
+        - name: end_height
+          in: query
+          required: false
+          description: The end height of the block range for events. Must be used together with `start_height`. This parameter is incompatible with `block_ids`.
+          schema:
+            $ref: '#/components/schemas/BlockHeight'
+          allowEmptyValue: false
+        - name: block_ids
+          in: query
+          style: form
+          required: false
+          explode: false
+          allowEmptyValue: false
+          schema:
+            type: array
+            items:
+              $ref: '#/components/schemas/Identifier'
+            minItems: 1
+            maxItems: 50
+            uniqueItems: true
+        - $ref: '#/components/parameters/selectParam'
```

Comment:
I would also recommend against having this endpoint, this was one that I purposely omitted when I was designing the API. There are a couple reasons:

* First, it is already possible to achieve a query for the same underlying data by querying the `/blocks` API by height range or ids, and then `expand`ing / `select`ing `execution_result.events`. If you look at the gRPC implementation for the `GetEventsBy...`  API's, this is exactly what happens underneath the hood.
* This API is not RESTful in nature, because events are not actually uniquely identifiable resources, and they only make sense within the context of a block or transaction. 

When we talk about feature parity, what we really mean is that we want to be able to do all the things we could do with the previous API using this API. This doesn't mean that we need to have all the same endpoints, or that things need to be queried the same way. Otherwise, it would have been very easy to simply design a JSON version of the gRPC API, or some other JSON RPC api. 

However, we specifically decided to design a REST API, and the existing gRPC API doesn't have RESTful semantics, so naturally there will be some differences in how they should be queried.

Reply:
I understand that feature parity is not having the same endpoints but the ability to achieve the same functionality. Although honestly, I overlooked this possibility of having events as part of blocks I still see a potential problem with the suggested approach for fetching events, that is, inability to query by the event type on the backend, I'm afraid that if our suggested option is for devs to fetch all events in big ranges of a block to only then filter them on the client-side it will generate a lot of data being sent over the network, don't you think? I imagine a lot of transactions in a block can be token transfers that have a lot of events and then fetching that for ranges of 50 blocks will amount to a lot of data. This is not necessarily an argument for my approach but more of a concern with the current suggestion. Adding this filter to block also doesn't really feel RESTy. What is your take on this?

Reply:
retaining this as per our [slack discussion](https://axiomzen.slack.com/archives/C02F8LFCCP8/p1639590253006600)